### ISSUE TITLE
How effective is my training so far?

### ISSUE THREAD

**Issue by michael-conrad** [Tomiinek/Multilingual_Text_to_Speech#16]
What type of loss_total number, etc, should I be looking for to verify that things seem to be training correctly?

I'm currently at step: 3.792k, 3 hours 12 minutes, total loss 0.2972

**Comment by michael-conrad** [comment]
![image](https://user-images.githubusercontent.com/5229492/89672430-9ee7ef80-d8b2-11ea-9fad-ec3d45a7b099.png)

![image](https://user-images.githubusercontent.com/5229492/89672453-ab6c4800-d8b2-11ea-8b48-e246bcc5b290.png)

![image](https://user-images.githubusercontent.com/5229492/89672510-c17a0880-d8b2-11ea-9cac-3203094d49f9.png)

**Comment by michael-conrad** [comment]
Predicted/forced, step 13

![image](https://user-images.githubusercontent.com/5229492/89672613-f4240100-d8b2-11ea-87c4-b63adfb863ee.png)

Predicted/generated, step 13

![image](https://user-images.githubusercontent.com/5229492/89672665-0bfb8500-d8b3-11ea-991d-eaf92dc0d743.png)

Target/eval, step 13

![image](https://user-images.githubusercontent.com/5229492/89672692-1a49a100-d8b3-11ea-8921-7c7c931c484f.png)

**Comment by Tomiinek** [comment]
The curves seem to be ok. You can expect the final MCD around 3 or 4.

**Comment by vince62s** [comment]
I am trying to replicate the generated_switching config with CSS10 + Comvoi clean
the only thing I changed is the batch size to 100 because I am using a RTX A6000.

MCD does not go below 5 could there be something wrong ?


![image](https://user-images.githubusercontent.com/15141326/143836782-87213a11-4664-4812-8dcf-a96d1811b437.png)

**Comment by Tomiinek** [comment]
Hi, I would also increase learning rate when increasing batch size, but the MCD values are fine IMHO. Give it more time.

Here are my MCDs, you are interested in the last column.
![image](https://user-images.githubusercontent.com/13215584/144051379-19f42977-0b6c-4450-bff7-8fca22937cc7.png)

**Comment by vince62s** [comment]
1) Is this the generated_switching or generated_training ? you don't have the similar grap for the eval set ?

2) one unrelated question. when generating the wav file with a speaker-id that has not been used in the WaveRNN training, can I expect a decent result or not at all. It seems to do something but unclear whether we can achieve something decent.
I am wondering whether it is just a matter of training WaveRNN on many more speakers or not.

thanks for your insights, great work btw.

**Comment by Tomiinek** [comment]
1. This is `generated_training`. I don't, but you can expect something 3-4ish I think.

2. You can check it out in the demo notebooks. For some voices it is ok, for some it is not. I am afraid that WaveRNN needs a lot of data per speaker to sound good ... but there is a more recent and SOTA workaround. I would suggest you using the pretrained vocoders from [espnet](https://github.com/espnet/espnet). They are multi-speaker and sound great but they also expect sharp spectrograms as input. To make outputs of my model sharper, you can replace the old convolutional postnet with something more sexy like a postnet based on normalizing flows ([here](https://github.com/keonlee9420/PortaSpeech/blob/main/model/postnet.py) is an open and probably working implementation)

**Comment by Tomiinek** [comment]
You can avoid the pain with vocoders by using espnet (it is for free) and replacing the postnet should be more or less copy & paste drop-in replacement and addition of one more loss term.
